eng/tools/ChangeValidation/SubmoduleValidation.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.Text;
+using static ValidateVmrChanges.Validate;
+
+namespace ValidateVmrChanges;
+
+internal class SubmoduleValidation
+{
+    // <summary>
+    // Verifies that the PR does not include any changes to submodule files
+    // </summary>
+    internal static List<ProcessingMessage> VerifySubModuleFileChanges(List<string> fileNames)
+    {
+        List<ProcessingMessage> messages = [];
+
+        var submoduleChanges = fileNames
+            .Where(f => f.Equals(".gitmodules", StringComparison.OrdinalIgnoreCase) || f.Contains("submodules/") || f.EndsWith("/.gitmodules"))
```

Comment:
This check is wrong, there's no rule about submodules being in the `submodules` directory:
- We should verify no submodules are added - which means changes to `.gitmodules` (so the first part of the condition)
- We should use DarcLib to read `source-manifest.json` and check the `submodules` list there to see if any files of the inlined submodules have been changed.